port/common/omrsysinfo_helpers.c

```diff
@@ -435,6 +435,35 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
 	desc->features[3] = CPUInfo[CPUID_EBX]; /* Structured Extended Feature Flags in EBX */
 	desc->features[4] = CPUInfo[CPUID_ECX]; /* Structured Extended Feature Flags in ECX */
 
+	if (portLibrary->sysinfo_processor_has_feature(portLibrary, desc, OMR_FEATURE_X86_OSXSAVE)) {
+		/* Record extended processor state features from XCR0 */
+		unsigned int leaf = 0;
+#if defined(OMR_OS_WINDOWS)
+		unsigned long long xcr0 = _xgetbv(leaf);
+#else /* defined(OMR_OS_WINDOWS) */
+		unsigned int eax;
+      unsigned int edx;
+		__asm__ __volatile__("xgetbv" : "=a"(eax), "=d"(edx) : "c"(leaf));
+		unsigned long long xcr0 = ((unsigned long long) edx << 32) | eax;
```

Comment:
- in c files, variables should be declared at the start of the scope since compilation errors are known to occur on some Windows compilers
- define `xcr0` once at the start, and then set it for the two different scenarios